rep-2004.rst

```diff
@@ -0,0 +1,689 @@
+Refer to the detailed description of the requirements following the Quality Level 1 section above for more information.
+
+Quality Level 4
+^^^^^^^^^^^^^^^
+
+These are demos, tutorials, or experiments.
+They don't have strict requirements, but are not excluded from having good documentation or tests.
+For example, this might be a tutorial package which is not intended for reuse but has excellent documentation because it serves primarily as an example to others.
+
+Package Requirements
+~~~~~~~~~~~~~~~~~~~~
+
+Requirements to be considered a 'Level 4' package:
+
+#. **Version Policy:**
+
+   #. No requirements, but having a policy is still recommended (e.g. ``semver``), even if the version is not yet stable (e.g. >= 1.0.0 for ``semver``)
+
+#. **Change Control Process:**
+
+   #. No explicit change control process required, but still recommended
+
+#. **Documentation:**
+
+   #. Must have a declared license or set of licenses
+   #. Must have a copyright statement in each source file
+
+#. **Testing:**
+
+   #. No explicit testing requirements, though covering some if not all of the features with tests is recommended
+
+#. **Dependencies:**
+
+   #. No restrictions
+
+#. **Platform Support:**
+
+   #. Must support all target platforms for the package's ecosystem.
+
+      * For ROS 2 this means supporting all tier 1 platforms, as defined in `REP-2000 <https://www.ros.org/reps/rep-2000.html#support-tiers>`_
+
+Any package that does not claim to be 'Level 3' or higher is automatically 'Level 4'.
+Refer to the detailed description of the requirements following the Quality Level 1 section above for more information.
+
+Quality Level 5
+^^^^^^^^^^^^^^^
+
+Packages in this category cannot even meet the simple 'Level 4' requirements, and for that reason should not be used.
+'Level 5' has no requirements, but we still strongly recommend all packages have a license and copyright statement, ideally in each file.
+
+Repository Organization
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Since these categories are applied on a per package basis, and since there may be more than one package per source repository, it's recommended that the strictest set of policies apply to the whole repository.
+This is recommended rather than trying to mix processes depending on which packages are changed in a given change request (pull request or merge request, etc.).
+If this is too onerous, then it's recommended to split lower quality packages out into a separate repository.
+
+Reference Implementation
+========================
+
+The `ROS 2 Developer Guide <https://index.ros.org/doc/ros2/Contributing/Developer-Guide/>`_ describes the policies we implement to achieve Quality Level 1 for ROS Core packages.
+
+The `rcutils package's quality declaration <https://github.com/ros2/rcutils/pull/202/files>`_ is one example of the conditions of this REP in practice on a non-trivial package.
+
+.. update link when that draft is merged
+
+Quality Declaration Template
```

Comment:
I suggest adding the item numbers to the template for quicker referencing.

Reply:
https://github.com/ros-infrastructure/rep/pull/244